langserver/server.go

```diff
@@ -79,9 +80,9 @@ func (s Server) Run() error {
 // CreateHeadlessServer creates a locked down server instance for the REST API.
 //
 // "locked down" in this case means, that the instance cannot send or receive any JSONRPC communication. Logging messages that the instance tries to send over JSONRPC are redirected to stderr.
-func CreateHeadlessServer(ctx context.Context, prometheusClient promClient.Client) (HeadlessServer, error) {
+func CreateHeadlessServer(ctx context.Context, prometheusClient promClient.Client, logger log.Logger) (HeadlessServer, error) {
```

Comment:
An idea to improve it a bit would be to have a struct that will contain all parameter of the method.
Like that if we have to add another parameter we will just have to add a new field in the struct indtead of changing the signature of the method that would cause a breaking change.

Of course it can be done in another PR if you like this idea :)

Reply:
> Of course it can be done in another PR if you like this idea. :)

Both parts of the above statement :)